src/scala/microsoft-spark-2-4/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -149,9 +171,12 @@ object DotnetRunner extends Logging {
           returnCode = closeDotnetProcess(process)
           closeBackend(dotnetBackend)
         }
-
         if (returnCode != 0) {
-          throw new SparkUserAppException(returnCode)
+          if(enableLogRedirection && stderrBuffer != null) {
```

Comment:
```suggestion
          if(stderrBuffer.isDefined) {
```

or you can try to use pattern matching.